src/frontier_util.c

```diff
@@ -2021,17 +2001,26 @@ static void CheckPartyIneligibility(void)
     {
         s32 i;
         s32 caughtBannedMons = 0;
-        s32 species = gFrontierBannedSpecies[0];
-        for (i = 0; species != 0xFFFF; i++, species = gFrontierBannedSpecies[i])
+        u16 species;
+
+        for (i = 0; i <= NATIONAL_DEX_COUNT; i++) // Loop to count the total number of caught banned species
         {
-            if (GetSetPokedexFlag(SpeciesToNationalPokedexNum(species), FLAG_GET_CAUGHT))
-                caughtBannedMons++;
+            species = NationalPokedexNumToSpecies(i);
+            if (gSpeciesInfo[species].isRestrictedLegendary || gSpeciesInfo[species].isMythical)
+            {
+                if (GetSetPokedexFlag(i, FLAG_GET_CAUGHT))
+                    caughtBannedMons++;
+            }
         }
         gStringVar1[0] = EOS;
         gSpecialVar_0x8004 = TRUE;
         count = 0;
-        for (i = 0; gFrontierBannedSpecies[i] != 0xFFFF; i++)
-            count = AppendCaughtBannedMonSpeciesName(gFrontierBannedSpecies[i], count, caughtBannedMons);
+        for (i = 0; i <= NATIONAL_DEX_COUNT; i++) // Loop to actually display the banned species' names (needs total count from the above)
+        {
+            species = NationalPokedexNumToSpecies(i);
+            if (gSpeciesInfo[species].isRestrictedLegendary || gSpeciesInfo[species].isMythical)
+                count = AppendCaughtBannedMonSpeciesName(species, count, caughtBannedMons);
+        }
```

Comment:
This is a good idea but unfortunately it takes to long. Could you bring back the hardcoded list for now? Maybe we have a better idea in the future.

Reply:
Alright sure, but only for this part since I think this is the one one where it matters.

Reply:
Alright, addressed

Reply:
If the table is still being used in the dialog display, modifying the flags will not make the dialog update, which make it less user-friendly.

Reply:
> If the table is still being used in the dialog display, modifying the flags will not make the dialog update, which make it less user-friendly.

So what do we want to do? Just sacrifice the speed for userability?